### ISSUE TITLE
build fails on freebsd 10 for uim-1.8.6

### ISSUE THREAD

**Issue by zhou0** [uim/uim#104]
gmake[1]: Entering directory '/usr/home/lzhou/github.com/uim/uim/releases/download/uim-1.8.6/uim-1.8.6/scm'
gmake  -C ../uim uim-module-manager && \
LIBUIM_SYSTEM_SCM_FILES=/usr/home/lzhou/github.com/uim/uim/releases/download/uim-1.8.6/uim-1.8.6/sigscheme/lib LIBUIM_SCM_FILES=/usr/home/lzhou/github.com/uim/uim/releases/download/uim-1.8.6/uim-1.8.6/scm LIBUIM_PLUGIN_LIB_DIR=/usr/home/lzhou/github.com/uim/uim/releases/download/uim-1.8.6/uim-1.8.6/uim/.libs UIM_DISABLE_NOTIFY=1 ../uim/uim-module-manager \
    --path ../scm --register        "skk" "tutcode" "byeoru" "latin" "elatin" "m17nlib" "xmload" "pyload" "viqr" "ipa-x-sampa" "look"  "ajax-ime" "social-ime" "google-cgiapi-jp" "baidu-olime-jp" 
gmake[2]: Entering directory '/usr/home/lzhou/github.com/uim/uim/releases/download/uim-1.8.6/uim-1.8.6/uim'
gmake[2]: Leaving directory '/usr/home/lzhou/github.com/uim/uim/releases/download/uim-1.8.6/uim-1.8.6/uim'
gmake[1]: *** [Makefile:776: installed-modules.scm] Floating point exception

**Comment by kou** [comment]
I couldn't reproduce it on FreeBSD 10.3-RELEASE-p2:

```text
gmake  -C ../uim uim-module-manager && \
LIBUIM_SYSTEM_SCM_FILES=/home/kou/work/c/uim-1.8.6/sigscheme/lib LIBUIM_SCM_FILES=/home/kou/work/c/uim-1.8.6/scm LIBUIM_PLUGIN_LIB_DIR=/home/kou/work/c/uim-1.8.6/uim/.libs UIM_DISABLE_NOTIFY=1 ../uim/uim-module-manager \
    --path ../scm --register        "skk" "tutcode" "byeoru" "latin" "elatin"  "xmload" "pyload" "viqr" "ipa-x-sampa" "look"  "ajax-ime" "social-ime" "google-cgiapi-jp" "baidu-olime-jp" 
gmake[1]: Entering directory '/usr/home/kou/work/c/uim-1.8.6/uim'
/bin/sh ../libtool --tag=CC   --mode=link cc  -g -O2 -pedantic -pipe -Wall -Wchar-subscripts -Wmissing-declarations -Wredundant-decls -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wcast-align -Wsign-compare -Wno-long-long -Wno-overlength-strings -no-cpp-precomp -DNDEBUG -Wno-redundant-decls -Wno-unused-function   -o uim-module-manager uim_module_manager-uim-module-manager.o libuim-scm.la libuim.la 
libtool: link: cc -g -O2 -pedantic -pipe -Wall -Wchar-subscripts -Wmissing-declarations -Wredundant-decls -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wcast-align -Wsign-compare -Wno-long-long -Wno-overlength-strings -no-cpp-precomp -DNDEBUG -Wno-redundant-decls -Wno-unused-function -o uim-module-manager uim_module_manager-uim-module-manager.o  ./.libs/libuim-scm.a ./.libs/libuim.a /home/kou/work/c/uim-1.8.6/uim/.libs/libuim-scm.a /home/kou/work/c/uim-1.8.6/sigscheme/libgcroots/.libs/libgcroots.a
gmake[1]: Leaving directory '/usr/home/kou/work/c/uim-1.8.6/uim'
Warning: Module ajax-ime is not a correct module.
Warning: Module social-ime is not a correct module.
Warning: Module google-cgiapi-jp is not a correct module.
Warning: Module baidu-olime-jp is not a correct module.
```

Can you look into how to reproduce it?

My configure command line: `./configure --prefix=/tmp/local`.

**Comment by tagoh** [comment]
dunno if this is similar what I saw but:
Making all in scm
make[1]: Entering directory '/builddir/build/BUILD/uim-1.8.6/scm'
make  -C ../uim uim-module-manager && \
LIBUIM_SYSTEM_SCM_FILES=/builddir/build/BUILD/uim-1.8.6/sigscheme/lib LIBUIM_SCM_FILES=/builddir/build/BUILD/uim-1.8.6/scm LIBUIM_PLUGIN_LIB_DIR=/builddir/build/BUILD/uim-1.8.6/uim/.libs UIM_DISABLE_NOTIFY=1 ../uim/uim-module-manager \
    --path ../scm --register "anthy" "anthy-utf8" "canna"     "skk" "tutcode" "byeoru" "latin" "elatin" "m17nlib" "xmload" "pyload" "viqr" "ipa-x-sampa" "look"  "ajax-ime" "social-ime" "google-cgiapi-jp" "baidu-olime-jp" "yahoo-jp"
make[2]: Entering directory '/builddir/build/BUILD/uim-1.8.6/uim'
make[2]: 'uim-module-manager' is up to date.
make[2]: Leaving directory '/builddir/build/BUILD/uim-1.8.6/uim'
make[1]: *** [Makefile:776: installed-modules.scm] Segmentation fault (core dumped)
make[1]: Leaving directory '/builddir/build/BUILD/uim-1.8.6/scm'
make: *** [Makefile:568: all-recursive] Error 1

This is reproducible with mock -rfedora-rawhide-x86_64 uim-1.8.6-16.fc28.src.rpm on Fedora.
and:
# LIBUIM_SYSTEM_SCM_FILES=/builddir/build/BUILD/uim-1.8.6/sigscheme/lib LIBUIM_SCM_FILES=/builddir/build/BUILD/uim-1.8.6/scm LIBUIM_PLUGIN_LIB_DIR=/builddir/build/BUILD/uim-1.8.6/uim/.libs UIM_DISABLE_NOTIFY=1 libtool --mode=execute gdb --args ./uim/uim-module-manager     --path ../scm --register "anthy" "anthy-utf8" "canna"     "skk" "tutcode" "byeoru" "latin" "elatin" "m17nlib" "xmload" "pyload" "viqr" "ipa-x-sampa" "look"  "ajax-ime" "social-ime" "google-cgiapi-jp" "baidu-olime-jp" "yahoo-jp"
GNU gdb (GDB) Fedora 8.1-8.fc28
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /builddir/build/BUILD/uim-1.8.6/uim/.libs/lt-uim-module-manager...done.
(gdb) r
Starting program: /builddir/build/BUILD/uim-1.8.6/uim/.libs/lt-uim-module-manager --path ../scm --register anthy anthy-utf8 canna skk tutcode byeoru latin elatin m17nlib xmload pyload viqr ipa-x-sampa look ajax-ime social-ime google-cgiapi-jp baidu-olime-jp yahoo-jp
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.27-3.fc28.x86_64
warning: Loadable section ".note.gnu.property" outside of ELF segments
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
warning: Loadable section ".note.gnu.property" outside of ELF segments

Program received signal SIGSEGV, Segmentation fault.
0x00007fffeb938d3e in mconv_rebind_buffer () from /lib64/libm17n.so.0
Missing separate debuginfos, use: dnf debuginfo-install anthy-9100h-34.fc28.x86_64 eb-4.4.3-10.fc28.x86_64 expat-2.2.5-3.fc28.x86_64 libdatrie-0.2.9-7.fc28.x86_64 libthai-0.1.27-2.fc28.x86_64 libxml2-2.9.7-2.fc28.x86_64 m17n-lib-1.8.0-1.fc28.x86_64 sqlite-libs-3.22.0-3.fc28.x86_64 xz-libs-5.2.3-7.fc28.x86_64 zlib-1.2.11-6.fc28.x86_64
(gdb) bt
#0  0x00007fffeb938d3e in mconv_rebind_buffer () from /lib64/libm17n.so.0
#1  0x00007fffebb559cc in convert_mtext2str (mtext=mtext@entry=0x555555ac9b50) at m17nlib.c:234
#2  0x00007fffebb5612b in get_input_method_short_desc (nth_=<optimized out>) at m17nlib.c:488
#3  0x00007ffff7bb9bad in call () at ../sigscheme/src/eval.c:415
#4  0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>, env@entry=140737183065424) at ../sigscheme/src/eval.c:499
#5  0x00007ffff7bba2cd in map_eval (args=args@entry=140737183069360, args_len=args_len@entry=0x7fffffffc4b0, env=140737183065424)
    at ../sigscheme/src/eval.c:573
#6  0x00007ffff7bb94d6 in call_closure () at ../sigscheme/src/eval.c:184
#7  0x00007ffff7bb98e9 in call () at ../sigscheme/src/eval.c:283
#8  0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>, env@entry=140737183065424) at ../sigscheme/src/eval.c:499
#9  0x00007ffff7bb9f08 in scm_s_begin (args=140737183068736, eval_state=<optimized out>) at ../sigscheme/src/syntax.c:830
#10 0x00007ffff7bb9b97 in call () at ../sigscheme/src/eval.c:417
#11 0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>, env@entry=30) at ../sigscheme/src/eval.c:499
#12 0x00007ffff7bc662d in scm_load_internal (filename=<optimized out>) at ../sigscheme/src/load.c:229
#13 0x00007ffff7791c31 in GCROOTS_call_with_gc_ready_stack (ctx=0x555555758390, proc=proc@entry=0x7ffff7bc6460 <scm_load_internal>,
    arg=arg@entry=0x555555906c00) at gcroots.c:114
#14 0x00007ffff7bc8c26 in scm_call_with_gc_ready_stack (arg=0x555555906c00, func=0x7ffff7bc6460 <scm_load_internal>)
    at ../sigscheme/src/storage-gc.c:376
#15 scm_load (filename=0x555555906c00 "/builddir/build/BUILD/uim-1.8.6/scm/m17nlib.scm") at ../sigscheme/src/load.c:195
#16 scm_require_internal (filename=0x555555906c00 "/builddir/build/BUILD/uim-1.8.6/scm/m17nlib.scm")
    at ../sigscheme/src/module-sscm-ext.c:229
#17 scm_p_require (filename=<optimized out>) at ../sigscheme/src/module-sscm-ext.c:245
#18 0x00007ffff7bb9bad in call () at ../sigscheme/src/eval.c:415
#19 0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>) at ../sigscheme/src/eval.c:499
#20 0x00007ffff7bbbcdb in guard_body (eval_state=0x7fffffffc9b0) at ../sigscheme/src/module-srfi34.c:448
#21 0x00007ffff7bb9bad in call () at ../sigscheme/src/eval.c:415
#22 0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>) at ../sigscheme/src/eval.c:499
#23 0x00007ffff7bba414 in scm_call (proc=proc@entry=140737182809522, args=args@entry=30) at ../sigscheme/src/eval.c:94
#24 0x00007ffff7bba594 in scm_dynamic_wind (before=<optimized out>, thunk=140737182809522, after=140737182809346)
    at ../sigscheme/src/continuation.c:215
#25 0x00007ffff7bb9b97 in call () at ../sigscheme/src/eval.c:417
#26 0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>) at ../sigscheme/src/eval.c:499
#27 0x00007ffff7bba414 in scm_call (proc=proc@entry=140737182809762, args=<optimized out>) at ../sigscheme/src/eval.c:94
#28 0x00007ffff7bba820 in scm_call_with_current_continuation (proc=140737182809762, eval_state=<optimized out>)
    at ../sigscheme/src/storage.c:321
#29 0x00007ffff7bc61d8 in scm_s_srfi34_guard (cond_catch=<optimized out>, body=<optimized out>, eval_state=0x7fffffffcd80)
    at ../sigscheme/src/module-srfi34.c:343
#30 0x00007ffff7bb9b7c in call () at ../sigscheme/src/eval.c:420
#31 0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>, env@entry=140737182814192) at ../sigscheme/src/eval.c:499
#32 0x00007ffff7bc503b in scm_s_or (args=<optimized out>, eval_state=0x7fffffffce80) at ../sigscheme/src/syntax.c:475
#33 0x00007ffff7bb9b97 in call () at ../sigscheme/src/eval.c:417
#34 0x00007ffff7bb9dea in scm_eval (obj=<optimized out>, env=<optimized out>, env@entry=140737182814192) at ../sigscheme/src/eval.c:499
#35 0x00007ffff7bc8344 in scm_s_let_internal (permitted=ScmCons, eval_state=0x7fffffffcfe0, body=140737354004128, bindings=30)
---Type <return> to continue, or q <return> to quit---q
(gdb) frame 1
#1  0x00007fffebb559cc in convert_mtext2str (mtext=mtext@entry=0x555555ac9b50) at m17nlib.c:234
234       mconv_rebind_buffer(converter, (unsigned char *)buffer_for_converter,
(gdb) p *converter
$1 = {lenient = 1735287139, last_block = 6646122, at_most = 4149771424, nchars = 32767, nbytes = 0, 
  result = MCONVERSION_RESULT_SUCCESS, status = {ptr = 0x21, dbl = 1.6304166312761136e-322, 
    c = "!\000\000\000\000\000\000\000zh\000\367\377\177\000\000\240|X\367\377\177\000\000\000\000\000\000\000\000\000\000\061\001", '\000' <repeats 18 times>, "\257\016\000\000\000\020\000\000\004\000\000\000\000\000\000\000\001", '\000' <repeats 186 times>}, 
  internal_info = 0x0}

converter->internal_info was somehow freed which looks weird to me.

In addition to that, this is the logs from valgrind:
==65== Invalid write of size 8
==65==    at 0x13257D3E: mconv_rebind_buffer (in /usr/lib64/libm17n.so.0.4.2)
==65==    by 0x1303F9CB: convert_mtext2str (m17nlib.c:234)
==65==    by 0x1304012A: get_input_method_short_desc (m17nlib.c:488)
==65==    by 0x4E46BAC: call (eval.c:415)
==65==    by 0x4E46DE9: scm_eval (eval.c:499)
==65==    by 0x4E472CC: map_eval (eval.c:573)
==65==    by 0x4E464D5: call_closure (eval.c:184)
==65==    by 0x4E468E8: call (eval.c:283)
==65==    by 0x4E46DE9: scm_eval (eval.c:499)
==65==    by 0x4E46F07: scm_s_begin (syntax.c:830)
==65==    by 0x4E46B96: call (eval.c:417)
==65==    by 0x4E46DE9: scm_eval (eval.c:499)
==65==  Address 0x110 is not stack'd, malloc'd or (recently) free'd

uim might broke something in m17n-lib. dunno. just FYI

**Comment by dai-vdr** [comment]
https://github.com/uim/uim/issues/104#issuecomment-367979860 fixed by #124
